### PR TITLE
Fix problems based on SonarLint in TorpedoStore.java

### PR DESCRIPTION
**Pull request by schauermannb**
- add "throw" to illegalArgumentException
- make random private class attribute
- change "=-" to "-="

### CODE REVIEW

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java [hunk 1]
```diff
@@ -13,6 +13,8 @@ public class TorpedoStore {
   private double FAILURE_RATE = 0.0; //NOSONAR
 
   private int torpedoCount = 0;
+  
+  private Random rand = new Random();
```
Comment (Zykor111) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java:17]: ok1

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java [hunk 2]
```diff
@@ -30,18 +32,18 @@ public TorpedoStore(int numberOfTorpedos){
 
   public boolean fire(int numberOfTorpedos){
     if(numberOfTorpedos < 1 || numberOfTorpedos > this.torpedoCount){
-      new IllegalArgumentException("numberOfTorpedos");
+      throw new IllegalArgumentException("numberOfTorpedos");
```
Comment (Zykor111) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java:35]: ok2
gj

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java [hunk 3]
```diff
@@ -30,18 +32,18 @@ public TorpedoStore(int numberOfTorpedos){
...
     // simulate random overheating of the launcher bay which prevents firing
-    Random generator = new Random();
-    double r = generator.nextDouble();
+    
+    double r = this.rand.nextDouble();
 
     if (r >= FAILURE_RATE) {
       // successful firing
-      this.torpedoCount =- numberOfTorpedos;
+      this.torpedoCount -= numberOfTorpedos;
```
Comment (Zykor111) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java:46]: ok3
ggwp

**Review by Zykor111 — CHANGES_REQUESTED** [review]
All in all, very good code, I am proud son

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java (hunk 1 above)
Comment (schauermannb) [reply src/main/java/hu/bme/mit/spaceship/TorpedoStore.java:17]: szerintem is ok :D

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java [hunk 4]
```diff
@@ -1,7 +1,7 @@
 package hu.bme.mit.spaceship;
 
 import java.util.Random;
-
+//EZ A VÁLTOZTATÁS
```
Comment (Zykor111) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java:4]: oké, jó

**Review by Zykor111 — APPROVED** [review]

**Pull request merged**